### PR TITLE
Add name sorting to ArchivableFilesFinder.

### PR DESCRIPTION
**Pull request by func0der**
This should address the issues discussed in https://github.com/composer/composer/discussions/10271

### CODE REVIEW

File: src/Composer/Package/Archiver/ArchivableFilesFinder.php [hunk 1]
```diff
@@ -84,7 +84,8 @@ public function __construct($sources, array $excludes, $ignoreFilters = false)
             ->in($sources)
             ->filter($filter)
             ->ignoreVCS(true)
-            ->ignoreDotFiles(false);
+            ->ignoreDotFiles(false)
+            ->sortByName(true);
```
Comment (stof) [inline src/Composer/Package/Archiver/ArchivableFilesFinder.php:88]: the natural sorting is not available in the Symfony version used by the composer phar (it was added in symfony/finder 4.2). So you should rather use the basic sorting by name instead, to keep the archiving consistent between `composer archive` and cases using composer as a library.

**Comment by func0der** [comment]
I have tested this change locally and it leads to zips that are identical across different machines.
Given, that you manually `touch` all files before ziping them.

**Review by naderman — APPROVED** [review]

**Comment by stof** [comment]
@naderman would be great that you approve the CI run (and you might want to re-evaluate the level of workflow approval)

**Comment by naderman** [comment]
@stof indeed missed that. @Seldaek want to change the approval requirements?

**Comment by Seldaek** [comment]
Thanks, makes sense.

**Pull request merged**